Sources/mParticle-MoEngage/mParticle_MoEngage.swift

```diff
@@ -0,0 +1,333 @@
+        // Removing the "$" from attribute as it's not required in MoEngage
+        if key.first == "$" {
+            key.removeFirst()
+        }
+        // Check for boolean value that mPartcle SDK has converted to NSNumber
+        if value is NSNumber, let number = value as? NSNumber, String(cString: number.objCType) == String(cString: (true as NSNumber).objCType) {
+            value = number.boolValue
+        }
+        MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: key, forAppID: workspaceId)
+        return execStatus(.success)
+    }
+
+    public func setUserAttribute(_ key: String, values: [Any]) -> MPKitExecStatus {
+        return self.setUserAttribute(key, value: values)
+    }
+
+    public func onIdentifyComplete(
+        _ user: FilteredMParticleUser, request: FilteredMPIdentityApiRequest
+    ) -> MPKitExecStatus {
+        return updateUser(user, request: request)
+    }
+
+    public func onLoginComplete(
+        _ user: FilteredMParticleUser, request: FilteredMPIdentityApiRequest
+    ) -> MPKitExecStatus {
+        return updateUser(user, request: request)
+    }
+
+    public func onLogoutComplete(
+        _ user: FilteredMParticleUser, request: FilteredMPIdentityApiRequest
+    ) -> MPKitExecStatus {
+        guard let settings = settings else { return execStatus(.requirementsNotMet) }
+        MoEngageSDKAnalytics.sharedInstance.resetUser(forAppID: settings.workspaceId)
+        return updateUser(user, request: request)
+    }
+
+    public func onModifyComplete(
+        _ user: FilteredMParticleUser, request: FilteredMPIdentityApiRequest
+    ) -> MPKitExecStatus {
+        return updateUser(user, request: request)
+    }
+
+    private func updateUser(
+        _ user: FilteredMParticleUser, request: FilteredMPIdentityApiRequest
+    ) -> MPKitExecStatus {
+        guard let settings = settings else { return execStatus(.requirementsNotMet) }
+        if let userId = request.userIdentities?[MPIdentity.alias.rawValue as NSNumber] {
+            MoEngageSDKAnalytics.sharedInstance.setAlias(userId, forAppID: settings.workspaceId)
+        } else if let userId = request.customerId {
+            MoEngageSDKAnalytics.sharedInstance.setUniqueID(userId, forAppID: settings.workspaceId)
+        }
+
+        if let email = request.email {
+            MoEngageSDKAnalytics.sharedInstance.setEmailID(email, forAppID: settings.workspaceId)
+        }
+
+        if let mobile = request.userIdentities?[MPIdentity.mobileNumber.rawValue as NSNumber] {
+            MoEngageSDKAnalytics.sharedInstance.setMobileNumber(mobile, forAppID: settings.workspaceId)
+        }
+
+        // WORKAROUND: for setATTStatus:withATTStatusTimestampMillis: callback not invoked
+        if let _ = request.userIdentities?[MPIdentity.iosAdvertiserId.rawValue as NSNumber] {
+            MoEngageSDKAnalytics.sharedInstance.enableIDFATracking(forAppID: settings.workspaceId)
```

Comment:
This would enable IDFATracking for any value of ATTStatus. Is that what you intended?

Reply:
@BrandonStalnaker during testing with older mParticle version we found a bug where IDFA tracking callback wasn't invoked correctly. After using the latest version, this is not needed anymore, hence removed it.